### ISSUE TITLE
`cargo build --release` fails due to missing `std` types

### ISSUE THREAD

**Issue by CodingAnarchy** [Jayshua/simple-cookie#2]
I'm not sure what exactly is happening here, but I'm trying to upgrade to `2.0.1` of this library from `1.0.0`, and running into some strange behavior. While `cargo build` works without issues, `cargo build --release` produces the following output:

```rust
error: cannot find macro `vec` in this scope
   --> /Users/mtanous/.cargo/registry/src/index.crates.io-6f17d22bba15001f/simple-cookie-2.0.1/src/lib.rs:271:19
    |
271 |     let mut output = vec![0; output_buffer_length];
    |                      ^^^

error: cannot find macro `vec` in this scope
   --> /Users/mtanous/.cargo/registry/src/index.crates.io-6f17d22bba15001f/simple-cookie-2.0.1/src/lib.rs:132:19
    |
132 |     let mut output = vec![0; encoded_buffer_size(value.as_ref().len()).expect("unreachable, len comes from a slice and no slice can be large enough to make this operation overflow")];
    |                      ^^^

error[E0412]: cannot find type `String` in this scope
   --> /Users/mtanous/.cargo/registry/src/index.crates.io-6f17d22bba15001f/simple-cookie-2.0.1/src/lib.rs:130:91
    |
130 | pub fn encode_cookie(key: SigningKey, name: impl AsRef<[u8]>, value: impl AsRef<[u8]>) -> String {
    |                                                                                           ^^^^^^ not found in this scope

error[E0412]: cannot find type `Vec` in this scope
   --> /Users/mtanous/.cargo/registry/src/index.crates.io-6f17d22bba15001f/simple-cookie-2.0.1/src/lib.rs:266:98
    |
266 | pub fn decode_cookie(key: SigningKey, name: impl AsRef<[u8]>, value: impl AsRef<[u8]>) -> Result<Vec<u8>, DecodeError> {
    |                                                                                                  ^^^ not found in this scope

error[E0433]: failed to resolve: use of undeclared type `String`
   --> /Users/mtanous/.cargo/registry/src/index.crates.io-6f17d22bba15001f/simple-cookie-2.0.1/src/lib.rs:134:2
    |
134 |     String::from_utf8(output).expect("unreachable, encode_cookie_advanced should always produce ascii data")
    |     ^^^^^^ use of undeclared type `String`
```

I believe the cause is this line:

```rust
#![cfg_attr(all(not(test), not(debug_assertions)), no_std)]
```

This appears to want to build the file as `no_std` for `release` builds, but that's almost never what is desired, IMO. I think what is actually needed is a `#![cfg(std)]` on the methods that require the standard types (e.g. `Vec`, `String`).

**Comment by Jayshua** [comment]
You're right. To be honest I'm not really sure how I messed this up - something must be really screwed up in the build config for my project that uses this library in no_std env causing it to link against std anyway. Definitely need to look into that too.

**Comment by Jayshua** [comment]
I believe ce48ddc081d6eb3711f27ee8d285364d80e2918f should fix this. If you could try that commit in your environment that would be super helpful!

**Comment by Jayshua** [comment]
This is fixed in version 2.0.2.
